### ISSUE TITLE
[Ch 08][Notebook][Minor] DataLoader shuffle과 MLM masking 시연 셀에 시드가 없어 실행마다 값이 바뀜

### ISSUE THREAD

**Issue by modesta-jihye-lee** [yoon-gu/neuqes-101#216]
## 대상

- **챕터**: Ch 08 (`08_tokenizer_datasets`)
- **산출물**: Notebook
  - `08_tokenizer_datasets/08_tokenizer_datasets.ipynb` cell031 (`DataLoader(..., shuffle=True)`)
  - `08_tokenizer_datasets/08_tokenizer_datasets.ipynb` cell040~041 (`DataCollatorForLanguageModeling(mlm_probability=0.15)` 시연)
- **기준**: 사용자가 2026-09-01 Colab T4에서 직접 실행한 노트북과 `executed/08_tokenizer_datasets.ipynb`를 셀 단위 대조

## 실제 결과

같은 `torch.__version__`(`2.11.0+cu128`)·같은 GPU(Tesla T4)에서 실행했는데도 다음 두 곳의 출력이 실행마다 완전히 다릅니다.

**cell031** — `DataLoader` 첫 배치의 라벨:

| | `executed/` | Colab 재실행 |
|---|---|---|
| `label values` | `[1, 3, 3, 3, 1, 1, 3, 3]` | `[1, 1, 0, 1, 2, 1, 1, 0]` |

**cell040~041** — MLM masking 통계·예시:

| | `executed/` | Colab 재실행 |
|---|---|---|
| `[MASK]` 비율 | 44/384 (11.5%) | 45/384 (11.7%) |
| masking된 단어(첫 샘플) | `truck barrington`, `.`, `i` 위치 등 | `stalk`, `been` 위치 등 (전혀 다른 자리) |

## 원인

노트북 전체에서 시드가 걸린 곳은 5,000건 서브샘플링(`ds["train"].shuffle(seed=42)`, cell009) 하나뿐입니다. 그 이후 등장하는

```python
loader = DataLoader(tokenized_torch, batch_size=8, shuffle=True)   # cell031, seed/generator 없음
```

```python
mlm_collator = DataCollatorForLanguageModeling(
    tokenizer=tokenizer,
    mlm=True,
    mlm_probability=0.15,   # cell040, 이것도 시드 없음
)
```

두 곳 모두 전역 시드나 `torch.Generator`를 지정하지 않아, PyTorch 기본 RNG 상태에 의존합니다. 라이브러리 버전과 무관하게(동일 버전에서도) 실행마다 값이 달라집니다 — sklearn 버전 드리프트(Ch05·06, 같은 검수에서 별도 확인)와는 원인이 다릅니다.

## 기대 결과

교육 목적상 "매번 다른 배치/마스킹이 나온다" 자체가 잘못된 것은 아니지만(오히려 `shuffle=True`·MLM 확률적 마스킹의 정상 동작), 노트북·`executed/`·WikiDocs에 박제된 값이 학습자 화면과 항상 어긋나는 결과를 낳습니다. 다음 중 하나 권장:

1. `DataLoader(..., generator=torch.Generator().manual_seed(42))`, `DataCollatorForLanguageModeling`에 시드 고정(가능한 방식으로) 후 값 확정
2. 또는 마크다운에 "실행마다 값이 달라집니다(의도된 동작)"라는 안내를 명시해, `executed/`에 박힌 특정 숫자·단어를 정답처럼 인용하지 않도록

CLAUDE.md 수치 표기 3단 규약 기준으로는 이 값들이 "확률적" 칸에 해당하므로, 페이지 산문이 있다면(현재는 코드 출력만 있어 해당 없음) 특정 값을 단정 인용하면 안 됩니다.

## 재현 절차

```bash
jupyter nbconvert --to notebook --execute 08_tokenizer_datasets/08_tokenizer_datasets.ipynb
# 두 번 이상 실행해 cell031·cell040~041 출력이 매번 바뀌는지 확인
```

## 심각도

**Minor** — 학습자가 잘못된 개념을 배우지는 않습니다(마스킹·셔플 개념 자체는 정확). 다만 노트북에 박제된 특정 값(`truck barrington`, `[1,3,3,3,...]`)을 근거로 설명하거나 채점 기준으로 삼으면 안 됩니다.

## 연결

챕터 검수 이슈 #32. 참고: Ch 15·16에서 등록된 `#175`·`#177`(모델 학습 시드 미고정)과 같은 유형이지만, 이쪽은 학습이 아니라 데이터 파이프라인 시연 셀이라 영향 범위가 작아 Major가 아닌 Minor로 분류.
